### ISSUE TITLE
Setting --shadow-file makes dmypy reload daemon(perf issue)

### ISSUE THREAD

**Issue by davidatbu** [python/mypy#9309]
This is a bug report.

* Minimal repro
```bash
$ touch test.py
$ cp test.py /tmp/test.py
$ dmypy run test.py
Daemon started
Success: no issues found in 22 source files
$ dmypy run test.py
Success: no issues found in 1 source file
$ dmypy run -- --shadow-file test.py /tmp/test.py test.py 
Restarting: configuration changed
Daemon stopped
Daemon started
Success: no issues found in 22 source files
```
* What is the actual behavior/output?
`dmypy` interprets setting `--shadow-file` as a "configuration change" and reloads the mypy daemon. This effectively counteracts the whole point of having a daemon since it takes a long time to reload the daemon.
* What is the behavior/output you expect?
`dmypy` should not interpret `--shadow-file` changes as configuration changes. It should use the existing daemon instead of reloading.
* What are the versions of mypy and Python you are using?
Python 3.8.2.
mypy 0.790+dev.2e9afec83a5214f4a0eaf57a96034230b8460379
The mypy install is from Github master as of thirty minutes ago.
* What are the mypy flags you are using? (For example --strict-optional)
The repro above describes this. There are no `mypy.ini` (or friends) files in the directory.

**Comment by davidatbu** [comment]
An additional note: Running `mypy` with `-v` shows that setting `--shadow-file` doesn't make mypy "invalidate it's cache".

Might it therefore be enough to modify the lines below to ignore the "shadow-file" option? (or perhaps create a new method on `Options` to achieve the same thing? 
https://github.com/python/mypy/blob/d0711bd33db2828529b062aabe41c1ea004546d1/mypy/dmypy_server.py#L307-L308

**Comment by gvanrossum** [comment]
That sounds reasonable. Maybe you can help by submitting a PR? I'd be happy
to review it.

**Comment by davidatbu** [comment]
> a PR?
Id' be happy to. Thanks for offering to review. Will ping you when I make the PR.

**Comment by gvanrossum** [comment]
Excellent!

**Comment by davidatbu** [comment]
Well, getting dmypy to use the already existing `options.select_options_affecting_cache()` was easy.

But now, dmypy just keeps whichever path it cached when the daemon was started(in other words, it actually uses the `--shadow-file` flag only from when the daemon started).

Fixing that seems to be a lot more involved than I originally thought, I don't think I'll be able to manage that in the near future.

**Comment by gvanrossum** [comment]
That's too bad, but I see it's more than you signed up for. :-)

Thanks for the bug report anyway.
